src/Core/Settings.h

```diff
@@ -80,9 +80,14 @@ class IColumn;
     M(Bool, use_uncompressed_cache, false, "Whether to use the cache of uncompressed blocks.", 0) \
     M(Bool, replace_running_query, false, "Whether the running request should be canceled with the same id as the new one.", 0) \
     M(UInt64, background_buffer_flush_schedule_pool_size, 16, "Number of threads performing background flush for tables with Buffer engine. Only has meaning at server startup.", 0) \
-    M(UInt64, background_pool_size, 16, "Number of threads performing background work for tables (for example, merging in merge tree). Only has meaning at server startup.", 0) \
+    M(UInt64, background_pool_size, 16, "Deprecated???", 0) \
+    M(UInt64, background_merges_count, 32, "Number of merges that could be performed in background. Has to be not less than `background_merges_pool_size`. Only has meaning at server startup.", 0) \
```

Comment:
`background_merges_count` -> `background_merges_concurrency`? count means well, counting

Reply:
I have some problems with naming... `background_merges_concurrency` is a good alternative naming for `background_pool_size`, isn't it? Not for tasks count?

Reply:
My suggestion is based on this https://techdifferences.com/difference-between-concurrency-and-parallelism.html

If i read the code correctly, then by `count` you actually mean concurrency.

And, by`background_pool_size` you mean _parallelism_. Technically, pool size is good name.

Reply:
I'm wrong about assuming pool_size means parallelism. It is also concurrency, just on a different level of abstraction...